src/internal_events/prometheus.rs

```diff
@@ -75,22 +73,41 @@ impl<'a> InternalEvent for PrometheusParseError<'a> {
 
     fn emit_metrics(&self) {
         counter!("parse_errors_total", 1);
+        counter!(
+            "component_errors_total", 1,
+            "stage" => "processing",
+            "error_type" => "parsing_failure",
```

Comment:
Should this be `parse_failed` to match similar metrics?

Reply:
Good catch, yes.